python/src/aiconfig/eval/metrics.py

```diff
@@ -147,3 +152,128 @@ async def _get_overall_positive_sentiment(output_datum: str) -> TextOverallPosit
         worst_value=TextOverallPositiveSentiment(pos=0.0, neg=1.0),
     ),
 )
+
+
+def make_structured_llm_metric(
+    chat_completion_create: common.CompletionTextToSerializedJSON,
+    eval_llm_name: str,
+    pydantic_basemodel_type: Type[T_BaseModel],
+    metric_name: str,
+    metric_description: str,
+    field_descriptions: dict[str, str] = {},
+) -> Metric[str]:
+    def _make_evaluation_fn(basemodel_type: Type[T_BaseModel]) -> EvaluationFunction[str]:
+        async def _evaluation_fn(output_datum: str) -> common.CustomMetricPydanticObject[T_BaseModel]:
+            resp = common.get_llm_structured_response(
+                input_text=output_datum,
+                chat_completion_create=chat_completion_create,
+                basemodel_type=basemodel_type,
+            )
+
+            # Intentional: unwrap and raise here to conform to the Metric interface.
+            match resp:
+                case Err(e):
+                    raise ValueError(f"Error getting structured response: {e}")
+                case Ok(data):
+                    return common.CustomMetricPydanticObject(data=data)
+
+        return _evaluation_fn
+
+    return Metric(
+        evaluation_fn=_make_evaluation_fn(pydantic_basemodel_type),
+        metric_metadata=EvaluationMetricMetadata(
+            name=metric_name,
+            description=metric_description,
+            extra_metadata=dict(
+                basemodel_type_name=pydantic_basemodel_type.__name__,
+                eval_llm_name=eval_llm_name,
+                field_descriptions_json=json.dumps(field_descriptions, sort_keys=True),
+            ),
+        ),
+    )
+
+
+def _make_openai_structured_llm_metric_helper(
```

Comment:
Seems strange to have openai-specific methods in this more general file -- can this be moved to eval/openai.py?

Reply:
Yeah, I can move this but it'll require shuffling some other stuff (like the definition of metric) to avoid circular imports. Even if it's just type imports which *could* be made to work cyclically, it tends to be cleaner to make the modules totally non-cyclic.

Reply:
We're probably going to end up generalizing this by moving to AIConfig anyway, so I'm going to leave this as-is for now so I don't do work that we know will be undone